server/plugin_release.go

```diff
@@ -230,170 +443,275 @@ func downloadAsset(assetURL, tempFolder string) (string, error) {
 	// Write the body to file
 	_, err = io.Copy(out, resp.Body)
 	if err != nil {
-		return "", err
+		return "", nil
 	}
-	LogInfo("Done downloading")
-	return filepath, nil
+
+	return pathToFile, nil
 }
 
-func copyFileToSigningServer(fileToCopy, tempFolder string) (string, error) {
-	LogInfo("Will copy the artifact to the signing server")
-	clientConfig, _ := auth.PrivateKey(Cfg.PluginSigningSSHUser, Cfg.PluginSigningSSHKeyPath, ssh.InsecureIgnoreHostKey())
-	host := fmt.Sprintf("%s:22", Cfg.PluginSigningSSHHost)
+// getPluginAsset polls till it finds the plugin tar file.
+func getPluginAsset(ctx context.Context, githubClient *github.Client, owner, repo, tag string) (*github.ReleaseAsset, error) {
+	LogInfo("Checking if the release asset is available")
 
-	clientConfig.Timeout = 30 * time.Minute
-	client := scp.NewClientWithTimeout(host, &clientConfig, 30*time.Minute)
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Minute)
+	defer cancel()
 
-	err := client.Connect()
-	if err != nil {
-		LogError("Couldn't establish a connection to the remote server. err=" + err.Error())
-		return "", err
-	}
+	for {
+		// Using timer to avoid memory leaks
+		timer := time.NewTimer(30 * time.Second)
+		defer timer.Stop()
 
-	f, _ := os.Open(fileToCopy)
-	defer client.Close()
-	defer f.Close()
+		release, _, err := githubClient.Repositories.GetReleaseByTag(ctx, owner, repo, tag)
+		if err != nil {
+			if err, ok := err.(*github.ErrorResponse); ok && err.Response.StatusCode == http.StatusNotFound {
+				LogInfo("get release by tag %s was not found, trying again shortly", tag)
+			} else {
+				return nil, err
+			}
+		}
 
-	filename := strings.Split(fileToCopy, "/")
-	serverPath := fmt.Sprintf("/tmp/%s", filename[len(filename)-1])
-	LogInfo(serverPath)
-	err = client.CopyFile(f, serverPath, "0777")
+		if release != nil {
+			var pluginAsset *github.ReleaseAsset
+			for i := range release.Assets {
+				assetName := release.Assets[i].GetName()
+				if strings.HasSuffix(assetName, ".tar.gz") {
+					if pluginAsset != nil {
```

Comment:
It's a bit hard to read. Maybe add a bool to indicate the asset was found

Reply:
wouldn't the bool be redundant? What if i name the variable `foundPluginAsset`?